bower.json

```diff
@@ -17,6 +17,10 @@
     "package.json"
   ],
   "dependencies": {
-    "purescript-prelude": "^3.0.0"
+    "purescript-prelude": "^3.0.0",
+    "purescript-foldable-traversable": "^3.3.0"
```

Comment:
Yes... Without running any benchmarks, my guess is that the `tailRecEff` machinery here would be significantly less efficient than this PR: https://github.com/purescript/purescript-tailrec/blob/master/src/Control/Monad/Rec/Class.purs

Generally speaking, having some kind of `traverseRec` seems like something potentially nice. But I feel like for `Eff` in particular, it makes sense to have something optimized.